test/functional/job/job_spec.lua

```diff
@@ -37,6 +37,32 @@ describe('jobs', function()
     eq({'notification', 'exit', {0, 0}}, next_msg())
   end)
 
+  it('changes to / when cwd provided', function()
+    nvim('command', "let g:job_opts.cwd = '/'")
+    nvim('command', "let j = jobstart('pwd', g:job_opts)")
+    eq({'notification', 'stdout', {0, {'/', ''}}}, next_msg())
+    eq({'notification', 'exit', {0, 0}}, next_msg())
+  end)
+
+  it('changes to random directory when cwd provided', function()
+    local dir = eval('tempname()')
+    mkdir(dir)
+    nvim('command', "let g:job_opts.cwd = '" .. dir .. "'")
+    nvim('command', "let j = jobstart('pwd', g:job_opts)")
+    eq({'notification', 'stdout', {0, {dir, ''}}}, next_msg())
+    eq({'notification', 'exit', {0, 0}}, next_msg())
+    rmdir(dir)
+  end)
+
+  it('fails to change to non-existent directory when provided', function()
+    local _, err = pcall(function()
+      nvim('command', "let g:job_opts.cwd = '/NONEXISTENT'")
+      nvim('command', "let j = jobstart('pwd', g:job_opts)")
+    end)
+    ok(string.find(err, "E475: Invalid argument: expected valid directory$") ~= nil)
+    eq({'notification', 'exit', {0, 1}}, next_msg())
```

Comment:
The build is hanging on this test. The exit message is not put on the queue because the error caused early return from `f_jobstart`. I guess no other message is on the queue, so `next_msg()` just waits indefinitely.

It should be fine to just remove this line.